cmd/ignition2rpm/main.go

```diff
@@ -0,0 +1,404 @@
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+)
+
+var Version string
+var excludePrefix string
+
+func main() {
+
+	var onceFrom string
+	var outputRPM string
+	var version bool
+	var packageCanOverride bool
+
+	flag.Set("logtostderr", "true")
+	flag.Set("stderrthreshold", "WARNING")
+	flag.Set("v", "2")
+
+	flag.StringVar(&excludePrefix, "exclude-prefix", "", "Exclude files with this prefix")
+	flag.StringVar(&onceFrom, "config", "", "Config file ign/machineconfig to read")
+	flag.StringVar(&outputRPM, "output", "", "Specity name of RPM file to write")
+	flag.BoolVar(&packageCanOverride, "can-override", false, "Include fake 'provides' for rpm-ostree (https://github.com/coreos/rpm-ostree/pull/3125)")
+	flag.BoolVar(&version, "version", false, "show the version ("+Version+")")
+
+	flag.Parse()
+	if version {
+		fmt.Printf("%s\n", Version)
+		os.Exit(0)
+	}
+
+	//just the filename so we can use it as the package name
+	fileName := strings.TrimSuffix(onceFrom, filepath.Ext(onceFrom))
+	fileName = filepath.Base(fileName)
+
+	var err error
+
+	// use the library functions to figure out what this is
+	configi, contentFrom, err := senseAndLoadOnceFrom(onceFrom)
+	if err != nil {
+		glog.Fatalf("Unable to decipher onceFrom config type: %s", err)
+	}
+
+	// I might use this later, but not yet
+	_ = contentFrom
+
+	//just so we can set the build host
+	hostname, _ := os.Hostname()
+
+	//specify some boilerplate metadata
+	packedRPM, err := rpmpack.NewRPM(rpmpack.RPMMetaData{
+		Name:        fileName,
+		Version:     "1",
+		Release:     "1",
+		Summary:     "A package packed from " + onceFrom,
+		Description: "This is a machine-packed RPM that has been packed by 'ignition2rpm'",
+		BuildTime:   time.Now(),
+		Packager:    "MCO ignition2rpm",
+		Vendor:      "RedHat OpenShift",
+		//Licence:     "",
+		BuildHost: hostname,
+	},
+	)
+
+	//Special sauce from: https://github.com/coreos/rpm-ostree/pull/3125
+	// add a fake provides that signals to rpm-ostree to let this package override files
+	if packageCanOverride {
+		packedRPM.RPMMetaData.Provides = append(packedRPM.RPMMetaData.Provides, &rpmpack.Relation{Name: "rpmostree(override)"})
+	}
+
+	if err != nil {
+		glog.Fatalf("Failed to create new RPM file: %s", err)
+	}
+
+	/*Adding prefixes alone is not enough to make it relocatable (nope, doesn't work)
+	magic numbers come from rpmtag.h
+	packedRPM.AddCustomTag(1098, rpmpack.EntryStringSlice([]string{"/etc/", "/var/", "usr"}))
+	*/
+
+	// have to process the envelope if it's machineconfig, otherwise we don't
+	switch c := configi.(type) {
+	case ign3types.Config:
+
+		//process the ignition into an RPM
+		err = Ign2Rpm(packedRPM, &c)
+		if err != nil {
+			glog.Fatalf("Unable to convert ignition to RPM payload: %v", err)
+		}
+	case mcfgv1.MachineConfig:
+
+		//shuck the ignition out of the machineconfig if need be
+		//TODO: fish the name out of the machineconfig for the metadata?
```

Comment:
The MachineConfig has a `.GetName()` method attached to it that can provide this info.

Reply:
Yeah, that TODO was more "I thought about using the machineconfig name, but I already made the RPM and set the metadata up there on line 68 based on the filename, so that might be a cool idea...later...if we keep developing it"   :smile:

Reply:
No worries! I mentioned it yesterday and somehow forgot to include it as one of my top-level comments.